docs/predictive-analytics/next-best-offer/index.md

```diff
@@ -0,0 +1,207 @@
+---
+seo:
+  title: Next Best Offer for Banking
+  description: This recipe demonstrates how to use ksqlDB to present relevant offers to banking customers.
+---
+
+# Next Best Offer for Banking
+
+Customers today are faced with never ending marketing messages from a variety of sources.  Often these messages are generic and don't have any consideration for the individual needs of the customer.  This one-size-fits-all approach leads to poor conversion rates.  A better approach is to tailor offerings that take into consideration the interests of the customer based on previous purchases or behavior.   This recipe demonstrates how to take existing customer information and provide a "Next Best Offer" to encourage sales and retain customers.
+
+## Step-by-step
+
+### Set up your environment
+
+Provision a Kafka cluster in [Confluent Cloud](https://www.confluent.io/confluent-cloud/tryfree/?utm_source=github&utm_medium=ksqldb_recipes&utm_campaign=next_best_offer).
+
+
+--8<-- "docs/shared/ccloud_setup.md"
+
+### Read the data in
+
+--8<-- "docs/shared/connect.md"
+
+```json
+--8<-- "docs/predictive-analytics/next-best-offer/source.json"
+```
+
+--8<-- "docs/shared/manual_insert.md"
+
+### Run the stream processing application
+
+This application will perform a series of joins between event streams and tables to calculate the next best offer for a banking customer based on their activity which should yield higher customer activity and satisfaction.
+
+--8<-- "docs/shared/ksqlb_processing_intro.md"
+
+```sql
+--8<-- "docs/predictive-analytics/next-best-offer/process.sql"
+```
+
+--8<-- "docs/shared/manual_cue.md"
+
+```sql
+--8<-- "docs/predictive-analytics/next-best-offer/manual.sql"
+```
+
+
+### Cleanup
+
+--8<-- "docs/shared/cleanup.md"
+
+## Explanation
+
+
+### Adding the lookup tables
+
+In the event stream you'll use for this recipe, each activity entry contains only the id for the customer. This is expected, as it's a common practice to have [normalized](https://en.wikipedia.org/wiki/Database_normalization) event streams.  But when it's time to analyze the data, it's important to have additional customer information to provide context for any analysts reviewing the results. You'll also need a table to use for the calculated offer.
+
+#### Creating the customer table and inserting records
+
+Create the table for customer information:
+
+```sql
+CREATE TABLE customers (
+    CUSTOMER_ID INTEGER PRIMARY KEY,
+    FIRST_NAME STRING,
```

Comment:
Per https://github.com/confluentinc/ksqldb-recipes/issues/146#issuecomment-1020484699, can this be updated to use `VARCHAR` instead of `STRING` (here and throughout the recipe)?

Reply:
@ybyzek - I've updated to use `VARCHAR` vs. `STRING`